packages/shopify-codemod/transforms/catch-undefined-to-catch-error.js

```diff
@@ -0,0 +1,17 @@
+export default function catchUndefinedToCatchError({source}, {jscodeshift: j}, {printOptions = {quote: 'single'}}) {
+  function hasNoErrorReferences(path) {
+    const usage = j(path).find(j.Identifier, {name: 'error'});
+
+    return usage.size() === 0;
+  }
+
+  return j(source)
+    .find(j.CatchClause, {
+      param: {
+        name: 'undefined',
+      },
+    })
+    .filter(hasNoErrorReferences)
```

Comment:
We're not consistent about this, but my preference is to reduce cognitive load by splitting more elaborate logic into functions with human readable names.  That way, someone can read through the main flow of the transform without having to unpack every single detail.

I didn't add `hasNoErrorReferences` to `utils` because:
- It deals with an edge case generated by `decaf`, and is unlikely to function correctly elsewhere
- It's best to let things propagate up to `utils` once we see it used in multiple locations; if something is never reused, it's more readable to keep it close to its caller